compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -1436,6 +1436,51 @@ public void exitListConstructorExpr(BallerinaParser.ListConstructorExprContext c
         this.pkgBuilder.addListConstructorExpression(getCurrentPos(ctx), getWS(ctx), argsAvailable);
     }
 
+    @Override
+    public void enterLetExpression(BallerinaParser.LetExpressionContext ctx) {
+        if (isInErrorState) {
+            return;
+        }
+
+        this.pkgBuilder.startLetVarList();
+    }
+
+    @Override
+    public void enterLetExpr(BallerinaParser.LetExprContext ctx) {
+        if (isInErrorState) {
+            return;
+        }
+
+        this.pkgBuilder.startLetVarList();
+    }
+
+    @Override
+    public void exitLetExpression(BallerinaParser.LetExpressionContext ctx) {
+        if (isInErrorState) {
+            return;
+        }
+
+        this.pkgBuilder.addLetExpression(getCurrentPos(ctx));
+    }
+
+    @Override
+    public void exitLetVarDecl(BallerinaParser.LetVarDeclContext ctx) {
```

Comment:
The grammar seems to allow annotations, but are we handling them?

https://github.com/ballerina-platform/ballerina-lang/pull/21209/files#diff-0e72b59e6b830c29e9c6161f3bb4177aR844

The following should fail since an annotation allowed only on const is being used with a let var, but doesn't.

```ballerina
int globalVar = 5;

const annotation v1 on source const;

public function main() {
    int b = let @v1 int x = 4 in 2 * x * globalVar;
}
```

Reply:
This is fixed:
The above code should print the following error:
`annotation 'v1' is not allowed on var`

Also added test cases to cover this. Please refer *tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java* on this PR